Civi/RemoteParticipant/RegistrationEventFactory.php

```diff
@@ -114,15 +114,30 @@ private function handleAdditionalParticipants(RegistrationEvent $registrationEve
         $additionalParticipantsProfile = \CRM_Remoteevent_RegistrationProfile::getRegistrationProfile(
           $event['event_remote_registration.remote_registration_additional_participants_profile']
         );
+        $currentAdditionalParticipantCount = 0;
         foreach ($additionalContactsData as $participantNo => &$contactData) {
+            $currentAdditionalParticipantCount++;
             $additionalParticipantsProfile->modifyContactData($contactData);
             $contactData['contact_type'] ??= 'Individual';
             $contactData['xcm_profile'] = $event['event_remote_registration.remote_registration_additional_participants_xcm_profile'];
             $additionalParticipantsData[$participantNo]['role_id'] ??= $this->getDefaultRoleId($event);
             $additionalParticipantsData[$participantNo]['event_id'] = $submissionData['event_id'];
 
+            // Check for waitlist.
+            if (
+                !empty($event['max_participants'])
+                && \CRM_Remoteevent_Registration::getRegistrationCount($event['id'])
+                    // Primary participant has not yet been created or its status is not counted, thus add 1.
```

Comment:
> Primary participant has not yet been created or its status is not counted

Otherwise it would be `update` instead of `registration`, right?

Reply:
Yes but this could also be the case for invitations. Maybe this isn't robust enough for all combinations of status, status classes, `is_counted` configurations, etc. but it's the same conditions as in the existing validation in https://github.com/systopia/de.systopia.remoteevent/blob/e730fe0eae7edcef1a9cb637d3c9c88205ff6a91/CRM/Remoteevent/RegistrationProfile.php#L208-L215

Reply:
Would it make sense to have the code for this calculation only once (and maybe document its limitations)?

Reply:
Yes it would, just haven't gotten around to doing it yet, so follow-up issue material unfortunately :disappointed:

Reply:
I think that at least a todo comment should be added then.